web_generator/lib/src/ast.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:code_builder/code_builder.dart';
+
+import 'interop_gen/generate.dart';
+
+sealed class Node {
+  abstract final String? name;
+  abstract final String id;
```

Comment:
Should this be an `ID`? Is there any reason to use a `String` that we have to parse later versus just creating the `ID` ahead of time?